packages/vite/src/node/server/middlewares/proxy.ts

```diff
@@ -51,6 +51,10 @@ export function proxyMiddleware(
     }
 
     proxy.on('error', (err, req, originalRes) => {
+      // originalRes can be falsy if the proxy itself errored
+      if (!originalRes) {
+        httpServer?.emit('error', err)
```

Comment:
Since `httpServer` can be `null` (For example, if Vite is running in middleware mode), I think it would be better to output the error here directly (like we do below).

Reply:
Thank you for pointing it out!

I will change it right away.